src/Symfony/Component/HttpFoundation/IpUtils.php

```diff
@@ -179,24 +179,46 @@ public static function checkIp6(string $requestIp, string $ip): bool
      * Anonymizes an IP/IPv6.
      *
      * Removes the last byte for v4 and the last 8 bytes for v6 IPs
+     *
+     * @param array{'v4'?: int<0, 4>, 'v6'?: int<0, 16>} $bytesToAnonymize
      */
-    public static function anonymize(string $ip): string
```

Comment:
what about adding two arguments instead?

Reply:
Both solution suits me, I don't have a strong opinion for either way. I implemented the two arguments way